pkg/apis/openebs.io/device/v1alpha1/devicenode.go

```diff
@@ -0,0 +1,69 @@
+/*
+ Copyright © 2021 The OpenEBS Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=devicenode
+
+// DeviceNode records information about all devices available
+// in a node. In general, the openebs node-agent creates the DeviceNode
+// object & periodically synchronizing the devices available in the node.
+// DeviceNode has an owner reference pointing to the corresponding node object.
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+type DeviceNode struct {
```

Comment:
I am wondering if we should stick to `BlockDevice` like we did in NDM . Lot of devices like FPGAs , etc are being extended to work with K8s. This might create confusion for end user . WDYT?